public/utils/savedObjectHelper.test.ts

```diff
@@ -0,0 +1,231 @@
+        expect(e.message).toStrictEqual('Visualization augmentation is disabled, please enable visualization:enablePluginAugmentation.');
+      }
+    });
+    it('createSavedObject with max associated objects to be 0', async () => {
+      setUIAugSettings(true, 0);
+      try {
+        await createSavedObjectAssociation('monitorId', embeddable);
+      } catch (e) {
+        expect(e.message).toStrictEqual('Cannot associate the plugin resource to the visualization due to the limit of the max\n' +
+          '          amount of associated plugin resources (0) with\n' +
+          '          0 associated to the visualization');
+      }
+    });
+  });
+
+  describe('getAlertingAugmentVisSavedObjs', function () {
+    setUIAugSettings();
+    const adPluginResource = {
+      type: 'ad detector',
+      id: 'detectorId',
+    };
+    it('getSavedObject and filter non-alerting objects', async () => {
+      const validObj1 = createMockSavedObject(pluginResource, fn)
+      const adObj1 = createMockSavedObject(adPluginResource, adFn, 'anomalyDetectionDashboards')
+      const visId = validObj1.visId;
+      setAugLoader([validObj1, adObj1]);
+      const receivedObject = await getAlertingAugmentVisSavedObjs(visId);
+      const expectedObject = {...validObj1, visReference: {
+          id: visId,
+          name: 'visName',
+          type: 'visualization'
+        }};
+      expect(receivedObject).toStrictEqual([expectedObject]);
+    });
+    it('getSavedObject and vis augmenter is disabled', async () => {
+      setUIAugSettings(false);
+      const validObj1 = createMockSavedObject(pluginResource, fn)
+      const visId = validObj1.visId;
+      setAugLoader([validObj1]);
+      try {
+        await getAlertingAugmentVisSavedObjs(visId);
+      } catch (e) {
+        expect(e.message).toStrictEqual('Visualization augmentation is disabled, please enable visualization:enablePluginAugmentation.');
+      }
+    });
+  });
+
+  describe('getAssociatedMonitorIds', function () {
+    it('getAssociatedMonitorIds', async () => {
+      setUIAugSettings();
+      const validObj1 = createMockSavedObject(pluginResource, fn)
+      const visId = validObj1.visId;
+      setAugLoader([validObj1]);
+      const receivedObject = await getAssociatedMonitorIds(visId);
+      expect(receivedObject).toStrictEqual([validObj1.pluginResource.id]);
+    });
+    it('getAssociatedMonitorIds and vis augmenter is disabled', async () => {
+      setUIAugSettings(false)
+      const validObj1 = createMockSavedObject(pluginResource, fn)
+      const visId = validObj1.visId;
+      setAugLoader([validObj1]);
+      try {
+        await getAssociatedMonitorIds(visId);
+      } catch (e) {
+        expect(e.message).toStrictEqual('Visualization augmentation is disabled, please enable visualization:enablePluginAugmentation.');
+      }
+    });
+  });
+
+  describe('deleteAlertingAugmentVisSavedObj', function () {
+    it('deleteAlertingAugmentVisSavedObj', async () => {
```

Comment:
Nitpick: It could be helpful for debugging purposes to make the `it` attribute a little more descriptive so it's easier to identify the specifics of the test case.